### DISCUSSION TITLE
How to use the pre-installed deno executable?

### DISCUSSION THREAD

**Discussion by yurivict** [quarto-dev/quarto-cli#10028] [Q&A]
### Description

package/scripts/common/utils.sh seems to  be downloading deno unconditonally, based on the architecture.

I already have deno installed. How to tell Quarto to use it?

**Comment by mcanouil** [comment]
Why do you want to use your installation of Deno?

Quarto uses things specific to a Deno version (Deno version: 1.41.0) that is why Quarto embeds Deno.
Same for Pandoc. (Gimp, gcloud, etc. also embeds their dependencies)

Asking Quarto to use your own Deno means you are almost certain to break Quarto.


You can use Quarto's Deno version: https://quarto.org/docs/projects/scripts.html#deno-scripts

**Comment by yurivict** [comment]
> Why do you want to use your installation of Deno?

Reason 1:  You don't provide the FreeBSD version of Deno
Reason 2: It is unsafe to just download binaries from some URLs. Binaries should be cryptographically fingerprinted, which you don't do.
Reason 3: We need to have Quarto packaged in order to be used as a dependency of RStudio. It is easier to have all dependencies packaged, than .... I don't even know what to do otherwise.

These are the reasons tthat wee need Quarto to accept external Deno and Pandoc binaries.
We can provide binaries of specific version expected by Quarto, this would still be better than downloading them during build.

**Reply by mcanouil** [reply]
Thanks for the clarification.

> Reason 3: We need to have Quarto packaged in order to be used as a dependency of RStudio

Quarto is provided in RStudio. Does it not work for FreeBSD?

**Reply by yurivict** [reply]
No. It fails to build with some cryptic error message.
Something about gmake not accepting the J flag.

But since Quarto is a separate software package, I think it would also be better to have a separate port/package for it.
I am trying to figure out how to do this.

**Reply by mcanouil** [reply]
I know there is `QUARTO_DENO` (path to the deno binary) that is used for/by installers such as the conda recipe.

The Deno version required is set in https://github.com/quarto-dev/quarto-cli/blob/0b9b55b0edce3c3b81f34c2a000bf5812a4981c9/configuration#L11-L12
